### PR TITLE
Improve RPM validation

### PR DESCRIPTION
**Pull request by atanasdinov**
- Closes https://github.com/suse-edge/edge-image-builder/issues/384

### CODE REVIEW

File: pkg/image/validation/os.go [hunk 1]
```diff
@@ -189,27 +206,24 @@ func validatePackages(os *image.OperatingSystem) []FailedValidation {
...
+	var repoURLs []string
 
-		if duplicates := findDuplicates(urlSlice); len(duplicates) > 0 {
-			duplicateValues := strings.Join(duplicates, ", ")
-			msg := fmt.Sprintf("The 'additionalRepos' field contains duplicate repos: %s", duplicateValues)
+	for _, repo := range os.Packages.AdditionalRepos {
+		if repo.URL == "" {
 			failures = append(failures, FailedValidation{
-				UserMessage: msg,
+				UserMessage: "The 'url' field is required for all entries under 'additionalRepos'.",
```
Comment (atanasdinov) [inline pkg/image/validation/os.go:214]: This is only reworded for consistency with all other functions within the module.

File: pkg/image/validation/os.go [hunk 2]
```diff
@@ -181,6 +182,22 @@ func validateSuma(os *image.OperatingSystem) []FailedValidation {
 func validatePackages(os *image.OperatingSystem) []FailedValidation {
 	var failures []FailedValidation
 
+	if len(os.Packages.PKGList) == 0 {
+		if len(os.Packages.AdditionalRepos) > 0 {
+			failures = append(failures, FailedValidation{
+				UserMessage: "The 'packageList' field is required if any entries are specified under 'additionalRepos'.",
```
Comment (ipetrov117) [inline pkg/image/validation/os.go:188]: Isn't this conflicting with the use-case where we have a RPM (in the `rpms` dir) that has some dependencies that should be retrieved from a third-party repository?

Then we would have a configuration where `AdditionalRepos` is defined without an entry in the package list.

Comment (atanasdinov) [reply pkg/image/validation/os.go]: Thanks, I didn't catch that at all. Added a comment to explain it for other developers.

**Review by ipetrov117 — APPROVED** [review]
Looks good.

**Pull request merged**